### PR TITLE
(EAI-575) [UI] Message Metadata Merging

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-575) [UI] Message Metadata Merging

## Changes

- Adds `updateMessageMetadata` method & uses it in place of `setMessageMetadata`

### CODE REVIEW

File: packages/mongodb-chatbot-ui/src/useConversation.tsx [hunk 1]
```diff
@@ -591,6 +595,22 @@ export function useConversation(params: UseConversationParams) {
...
+    async ({ messageId, metadata }) => {
+      if (!state.conversationId) {
+        console.error(`Cannot updateMessageMetadata without a conversationId`);
+        return;
+      }
+      dispatch({
+        type: "setMessageMetadata",
+        messageId,
+        metadata: {
+          ...state.messages.find((m) => m.id === messageId)?.metadata,
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/useConversation.tsx:608]:
```suggestion
          ...(state.messages.find((m) => m.id === messageId)?.metadata ?? {}),
```

i think you need this so spread operator doesn't break on undefined

File: packages/mongodb-chatbot-ui/src/useConversation.tsx [hunk 2]
```diff
@@ -591,6 +595,22 @@ export function useConversation(params: UseConversationParams) {
...
+          ...metadata,
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/useConversation.tsx:609]: cna metadata be undefined? if so, i think prefer nullish spreading

```suggestion
          ...(metadata ?? {}),
```

**Review by mongodben — CHANGES_REQUESTED** [review]
see notes. also build failing.

also, i think this file really ought to be tested given the amount of core app logic in it.

File: packages/mongodb-chatbot-ui/src/useConversation.tsx (hunk 1 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/useConversation.tsx:608]: Spread operator supports undefined

`{ ...undefined } => {}`

File: packages/mongodb-chatbot-ui/src/useConversation.tsx (hunk 2 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/useConversation.tsx:609]: We validate that the event is well-formed & the field exists in the `addMessageStreaming` implementation so this won't be undefined. Also spread operator supports undefined.

**Comment by nlarew** [comment]
> this file really ought to be tested

Fair enough, needs some setup / probably investigation into config stuff

https://jira.mongodb.org/browse/EAI-576

File: packages/mongodb-chatbot-ui/src/useConversation.tsx (hunk 1 above)
Comment (mongodben) [reply packages/mongodb-chatbot-ui/src/useConversation.tsx:608]: noted! i must have been hallucinating...

**Pull request merged**